nengo/cache.py

```diff
@@ -321,6 +321,11 @@ def shrink(self, limit=None):
             Maximum size of the cache in bytes.
         """
         if self.readonly:
+            warnings.warn("Cannot shrink a readonly cache.")
```

Comment:
Hmm, what do you think about changing it to a `logger.info` in that case? The benefit of the warning is that it'll only go off once, but you're right that it's not something that the user necessarily needs to see.